src/services/filters.js

```diff
@@ -317,3 +317,107 @@ ngeo.trustHtmlFilter = function($sce) {
 };
 
 ngeo.module.filter('ngeoTrustHtml', ngeo.trustHtmlFilter);
+
+
+/**
+ * A filter used to format a time duration in seconds into a more
+ * readable form.
+ * Only the two largest units will be shown.
+ *
+ * Examples:
+ *      {{42 | ngeoDuration}} => 42 seconds
+ *      {{132 | ngeoDuration}} => 2 minutes 12 seconds
+ *      {{3910 | ngeoDuration}} => 1 hour 5 minutes
+ *        -> Note: the remaining 10 seconds will be dropped
+ *
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @return {ngeox.duration} Function used to format a time duration
+ *  in seconds into a string.
+ * @ngInject
+ * @ngdoc filter
+ * @ngname ngeoDuration
+ */
+ngeo.durationFilter = function(gettextCatalog) {
+  const gettextCatalog_ = gettextCatalog;
+
+  // time unit enum
+  const TimeUnits = Object.freeze({
+    SECONDS: Symbol('seconds'),
+    MINUTES: Symbol('minutes'),
+    HOURS: Symbol('hours'),
+    DAYS: Symbol('days')
+  });
+
+  /**
+   * @param {number} amount Amount of time.
+   * @param {symbol} unit Unit of time.
+   * @return {string} formatted and translated string
+   */
+  const pluralize = function(amount, unit) {
+    let formattedUnit = '';
+    switch (unit) {
+      case TimeUnits.SECONDS:
+        formattedUnit = amount > 1 ? 'seconds' : 'second';
+        break;
+      case TimeUnits.MINUTES:
+        formattedUnit = amount > 1 ? 'minutes' : 'minute';
+        break;
+      case TimeUnits.HOURS:
+        formattedUnit = amount > 1 ? 'hours' : 'hour';
+        break;
+      case TimeUnits.DAYS:
+        formattedUnit = amount > 1 ? 'days' : 'day';
+        break;
+      default:
+        break;
+    }
+    return `${amount} ${gettextCatalog_.getString(formattedUnit)}`;
```

Comment:
You should put the gettextCatalog_.getString directly on each string to make the string collection working ...

and it should probably be named gettextCatalog without ending _